src/WebJobs.Extensions.DurableTask/HttpApiHandler.cs

```diff
@@ -899,6 +941,48 @@ private async Task<HttpResponseMessage> HandleStartOrchestratorRequestAsync(
             }
         }
 
+        private static string GetHeaderValueFromHeaders(string header, HttpRequestHeaders headers)
+        {
+            if (headers.TryGetValues(header, out IEnumerable<string> values))
+            {
+                return values.FirstOrDefault();
+            }
+
+            return null;
+        }
+
+        internal static Activity StartActivityForNewOrchestration(ExecutionStartedEvent startEvent, ActivityContext parentTraceContext)
+        {
+            // Create the Activity Source for the WebJobs extension
+            ActivitySource activitySource = new ActivitySource("WebJobs.Extensions.DurableTask");
+
+            // Start the new activity to represent scheduling the orchestration
+            Activity newActivity = activitySource.CreateActivity(
+                name: Schema.SpanNames.CreateOrchestration(startEvent.Name, startEvent.Version),
+                kind: ActivityKind.Producer,
+                parentContext: parentTraceContext);
+
+            newActivity.Start();

```

Comment:
Potential null-ref exception here? In the line below, I see you're checking for null but you're not checking it for this line.